From 96a6e59b817ae30affd8ae4c274f8773766d77e5 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Mon, 25 Nov 2024 04:00:05 +0200
Subject: [PATCH 10/15] Adapt to modern Qt6 QML import path conventions

Let's adopt QTP0001 already and move all per-application resources under
qrc:/qt/qml/AsteroidApp/
---
 cmake/AsteroidCMakeSettings.cmake  | 3 ++-
 src/app/AsteroidAppConfig.cmake.in | 2 ++
 src/app/asteroidapp.cpp            | 2 +-
 src/app/asteroidapp.h              | 2 +-
 4 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/cmake/AsteroidCMakeSettings.cmake b/cmake/AsteroidCMakeSettings.cmake
index fe11c06..6e28ad5 100644
--- a/cmake/AsteroidCMakeSettings.cmake
+++ b/cmake/AsteroidCMakeSettings.cmake
@@ -34,5 +34,6 @@ if (NOT ASTEROID_SKIP_BUILD_SETTINGS)
 	set(INSTALL_QML_IMPORT_DIR "${CMAKE_INSTALL_FULL_LIBDIR}/qml" 
 		CACHE PATH "Custom QML import installation directory")
 
-	set(QT_MIN_VERSION "5.12.0")
+	# Ensure availability of QTP0001 (https://doc.qt.io/qt-6/qt-cmake-policy-qtp0001.html)
+	set(QT_MIN_VERSION "6.5.0")
 endif()
diff --git a/src/app/AsteroidAppConfig.cmake.in b/src/app/AsteroidAppConfig.cmake.in
index c1afb72..012559e 100644
--- a/src/app/AsteroidAppConfig.cmake.in
+++ b/src/app/AsteroidAppConfig.cmake.in
@@ -8,3 +8,5 @@ set(ASTEROID_MODULE_PATH "@PACKAGE_ASTEROID_MODULES_INSTALL_DIR@")
 
 include("${CMAKE_CURRENT_LIST_DIR}/AsteroidAppTargets.cmake")
 @PACKAGE_INCLUDE_QCHTARGETS@
+
+qt_policy(SET QTP0001 NEW)
diff --git a/src/app/asteroidapp.cpp b/src/app/asteroidapp.cpp
index 13ae7e8..68f8db5 100644
--- a/src/app/asteroidapp.cpp
+++ b/src/app/asteroidapp.cpp
@@ -96,7 +96,7 @@ namespace AsteroidApp {
     {
         QScopedPointer<QGuiApplication> app(AsteroidApp::application(argc, argv));
         QScopedPointer<QQuickView> view(AsteroidApp::createView());
-        view->setSource(QUrl("qrc:/main.qml"));
+        view->setSource(QUrl("qrc:/qt/qml/asteroidapp/main.qml"));
         view->resize(app->primaryScreen()->size());
         view->show();
         return app->exec();
diff --git a/src/app/asteroidapp.h b/src/app/asteroidapp.h
index 077c39b..4f07052 100644
--- a/src/app/asteroidapp.h
+++ b/src/app/asteroidapp.h
@@ -43,7 +43,7 @@ namespace AsteroidApp {
     ASTEROIDAPP_EXPORT QGuiApplication *application(int &argc, char **argv);
     ASTEROIDAPP_EXPORT QQuickView *createView();
 
-    // Very simple interface: Uses "qrc:/main.qml" as QML entry point
+    // Very simple interface: Uses "qrc:/qt/qml/asteroidapp/main.qml" as QML entry point
     ASTEROIDAPP_EXPORT int main(int &argc, char **argv);
 };
 
-- 
2.47.0

